test/Game.ts

```diff
@@ -36,47 +38,79 @@ describe("Game ", function () {
       expect(await game.count()).to.equal(1)
     });
 
-    it("dao is not game.owner()", async function () {
-      expect(await game.owner()).to.not.equal(dao.address)
+    it("when owner is msg.sender, dao is not game impersonator", async function () {
+      expect(await game.impersonator()).to.not.equal(dao.address)
+    });
+
+    it("when dao is msg.sender, dao is not game impersonator", async function () {
+      expect(await daoGame.impersonator()).to.not.equal(dao.address)
+    });
+
+    it("dao is daoGame msg.sender", async function () {
+      expect(await daoGame.whoami()).to.equal(dao.address)
+    });
+
+    it("dao cannot impersonate randomPlayer b/c not curr impersonator", async function () {
+      await expect(daoGame.impersonateMe(randomPlayer.address)).to.be.revertedWith('only impersonator can update impersonator');
+    });
+
+    it("dao cannot increment b/c not curr impersonator", async function () {
+      await expect(daoGame.increment()).to.be.revertedWith('only impersonator can update count');
     });
 
-    it("owner can set setOwner", async function () {
-      await game.setOwner(dao.address);
-      expect(await game.owner()).to.equal(dao.address);
+    it("owner can allow dao to impersonate", async function () {
+      await game.impersonateMe(dao.address);
+      expect(await game.impersonator()).to.equal(dao.address);
     });
 
-    it("dao cannot set owner", async function () {
-      expect(await daoGame.whoami()).to.equal(dao.address);
-      await expect(daoGame.setOwner(dao.address)).to.be.revertedWith('_getImpersonator is not owner');
+    it("owner can impersonate dao, then randomPlayer", async function () {
```

Comment:
I think this verbage feels not correct. maybe
```suggestion
    it("owner can set dao as impersonator, then randomPlayer", async function () {
```